core/templates/dev/head/pages/exploration_player/ExplorationStateService.js

```diff
@@ -0,0 +1,35 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service that maintains state of the current exploration.
+ */
+
+oppia.factory('explorationStateService', [function() {
+  var _exploration = null;
```

Comment:
Drop leading underscore; we no longer do that. In setExploration() you can call the argument something like newExploration.